File: tests/conformance/fixtures/go/05-matrix.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("test").Run("go test ./...").
		Matrix("os", "linux", "darwin").
		Matrix("arch", "amd64", "arm64")

	p.Emit()
}

File: tests/conformance/fixtures/elixir/05-matrix.exs
use Sykli

pipeline do
  task "test" do
    run "go test ./..."
    matrix "os", ["linux", "darwin"]
    matrix "arch", ["amd64", "arm64"]
  end
end
|> Sykli.Emitter.validate!()
|> Sykli.Emitter.to_json()
|> IO.puts()
